### PR TITLE
Added OnError event to MetaObjectLock

### LINKED ISSUE #1584: Improve observability of Lease lock issues
### Describe the feature
When using the LeaderElector class to carry out the election pattern amongst pods, you can choose the ILock implementation. LeaseLock.cs uses the client to perform CRUD operations against a Lease resource. However all HTTP operation errors are swallowed, making debugging difficult.

A way of catching these errors should be exposed.

### Is your feature related to a problem?
When I deployed my pod using the kubernetes client, I forgot to give my service account suitable permissions. I gave it get, create, patch and delete but not update. Took me a long time to figure out why it was not working, because there is no observability in the code.

### Describe the requested feature
Add OnError event hook to LeaseLock, which fires when HTTP operation errors happen.

### Describe alternatives you've considered
Stepping through the code rather than observing behaviour by logs or traces.

### Additional Context
N/A

### PR DESCRIPTION
**Pull request by fh-tmccurdy**
Exposes HTTP operation exceptions for logging or tracing.

Fixes #1584

### CODE REVIEW

**Comment by brendandburns** [comment]
Looks ok to me. I'll wait for input from @tg123 

You'll need to sign our CLA.

File: src/KubernetesClient/LeaderElection/ResourceLock/MetaObjectLock.cs [hunk 1]
```diff
@@ -9,6 +9,11 @@ public abstract class MetaObjectLock<T> : ILock
         private readonly string identity;
         private T metaObjCache;
 
+        /// <summary>
+        /// OnError is called when there is a http operation error.
+        /// </summary>
+        public event Action<HttpOperationException> OnError;
```
[the remark this answers is not in the document]
Comment (fh-tmccurdy) [reply src/KubernetesClient/LeaderElection/ResourceLock/MetaObjectLock.cs:15]: I did consider it, so happy to use it if everyone agrees. I was just trying to match LeaderElectors event name.

[the remark this answers is not in the document]
Comment (tg123) [reply src/KubernetesClient/LeaderElection/ResourceLock/MetaObjectLock.cs:15]: what about `Action<Exception> OnError`
to align with all other OnError in this libarary

**Review by tg123 — APPROVED** [review]
/LGTM

**Pull request merged**